mautrix/client/api/modules/media_repository.py

```diff
@@ -121,21 +117,36 @@ async def upload_media(
         if filename:
             query["filename"] = filename
 
+        uploadUrl = None
+
         if async_upload:
             if mxc:
                 raise ValueError("async_upload and mxc can't be provided simultaneously")
-            mxc = await self.unstable_create_mxc()
+            createResponse = await self.unstable_create_mxc()
+            mxc = createResponse["content_uri"]
+            uploadUrl = createResponse.get("upload_url")
```

Comment:
Class fields should be accessed directly

```suggestion
            create_response = await self.unstable_create_mxc()
            mxc = create_response.content_uri
            upload_url = create_response.upload_url
```